iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -226,6 +227,90 @@ public override async Task<Message> ReceiveAsync(TimeoutHelper timeoutHelper)
             }
         }
 
+        public override async Task EnableReceiveMessageAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (Logging.IsEnabled)
+                {
+                    Logging.Enter(this, cancellationToken, nameof(EnableReceiveMessageAsync));
+                }
+
```

Comment:
I see inconsistent behavior when it comes to 
`cancellationToken.ThrowIfCancellationRequested();`
In methods that take cancellationToken as a parameter.
I think we have to have that statement at the beginning and in between retry attempts

Reply:
after looking at it, it does seem like all the methods we call already throw if cancellation is requested, so we may not need it after all.

Reply:
right.